src/JustEat.StatsD/Buffered/BufferBasedStatsDPublisher.cs

```diff
@@ -43,7 +43,7 @@ public void Timing(long duration, double sampleRate, string bucket)
 
         private void SendMessage(double sampleRate, in StatsDMessage msg)
         {
-            bool shouldSendMessage = sampleRate >= DefaultSampleRate || sampleRate > Random.NextDouble();
+            bool shouldSendMessage = (sampleRate >= DefaultSampleRate || sampleRate > Random.NextDouble()) && msg.StatBucket != null;
```

Comment:
Is this a place to use `! String.IsNullOrEmpty(msg.StatBucket)` ?

Reply:
If we want to. I was using null to try and force the exception branch of the code but couldn’t, it was just pointlessly sending a metric, so just added this in instead.